pints/_optimisers/_irpropmin.py

```diff
@@ -178,7 +238,50 @@ def tell(self, reply):
         self._current_df = dfx
 
         # Take step in direction indicated by current gradient
-        self._proposed = self._current - self._step_size * np.sign(dfx)
+        p = self._current - self._step_size * np.sign(dfx)
+
+        # Allow boundaries to reduce step size
+        if self._lower is not None:
```

Comment:
See comment above about making more clear that this is only used for rectangular boundaries